scripts/widgets/mz-search-pagination-dd.js

```diff
@@ -0,0 +1,168 @@
+                });
+            }
+
+            $(document).on('click', '.lastPage', function (evt) {
+                evt.stopImmediatePropagation();
+                paginationLoad(lastPageIndex - 1);
+                $('.displayPageCount').text((lastPageIndex + ' of ' + lastPageIndex));
+                pageIndex = lastPageIndex - 1; 
+            });
+
+            $(document).on('click', '.firstPage', function (evt) {
+                evt.stopImmediatePropagation();
+                paginationLoad(0);
+                pageIndex = 0;
+                $('.displayPageCount').text((pageIndex+1 + ' of ' + lastPageIndex));
+            });
+
+            $(document).on('click', '.prevPage', function (evt) {
+                evt.stopImmediatePropagation();
+                if (pageIndex >= 1) {
+                    $('.displayPageCount').text((pageIndex + ' of ' + lastPageIndex));
+                    pageIndex--;
+                    paginationLoad(pageIndex);
+                }
+                            
+                
+            });
+
+            $(document).on('click', '.nextPage', function (evt) {
+                evt.stopImmediatePropagation();
+                pageIndex++;
+                if (pageIndex <= lastPageIndex-1) {
+                    paginationLoad(pageIndex);
+                    $('.displayPageCount').text((pageIndex + 1 + ' of ' + lastPageIndex));
+                }     
+            });
+
+            $('<input class="mz-dd-search" type="text" placeholder="' + options.placeHolder + '" /><a style="text-decoration: none" class="mz-dd-open" href="javascript:void(0)">&#x2B9F</a>').prependTo(self.selector);
+            $(self.selector).wrapAll("<div  style='margin: 10px 0; position: relative; width: 300px; border: 1px solid #666;' />");
+            $('.mz-pagination-url').wrapAll("<div class='pagination_btn'></div>");
+            $('.pagination_btn, .listData').wrapAll("<div class='dropdown_data' style='display:none;'></div>");
+            
+            $(document).on("click", '.listData > a', function () {
+                $(self.selector + '> .mz-dd-search').val('');
+                $(self.selector + '> .mz-dd-search').val($(this).text());
+                $('.listData > a').removeClass("mz-dd-active");
+                $(this).addClass("mz-dd-active");
+                $('.dropdown_data').hide();
+                $(self.selector + '> .mz-dd-search').attr('data-id', $(this).data('id'));
+                filteredVal = false;                
+            });
+                    
+            $(document).on('click focus', '.mz-dd-search', function () {
+                $('.dropdown_data').show();
+            });
+
+            // API call for dropdown search
+            $(self.selector + '> .mz-dd-search').keyup(function (e) {
+                filteredVal = true;
+                $(self.selector + '> .mz-dd-search').removeAttr("data-id"); 
+                if (e.which !== 40 && e.which !== 38) {
```

Comment:
No magic integers. Key codes should be defined as constants at the top of the file.

Reply:
Done